code/obj/nuclearreactor/nuclearreactor.dm

```diff
@@ -626,35 +622,46 @@
 		src.power = power
 
 	on_pre_hit(atom/hit, angle, var/obj/projectile/O)
-		. = FALSE //default to doing normal hit behaviour
-
 		if(isintangible(hit) || isobserver(hit))
-			return TRUE
+			return TRUE //don't irradiate ghosts
 
 		var/multiplier = istype(hit,/turf/simulated/wall/auto/reinforced) ? 5 : 10
-		if((hit.material && !prob(hit.material.getProperty("density")*multiplier)) || (isnull(hit.material) && prob(5*multiplier)))
-			O.initial_power /= 2 //initial power because projectile.collide() uses it to calculate power, rather than direct use of .power
-			if(O.initial_power < 1)
-				O.initial_power = 0
-			return TRUE //don't hit this, lose power and pass through it
+		var/density = (hit.material ? hit.material.getProperty("density") : 3) //3 is default density
 
-		if(hit.material)
-			if(prob(hit.material.getProperty("hard")*10))
+		//first are we colliding with this or ignoring it?
+		if(prob(density*multiplier))
+			//we hit it! now decide what that hit means
+			//first, reflection
+			if(hit.material && prob(hit.material.getProperty("hard")*10))
 				//reflect
-				shoot_reflected_bounce(O, hit)
-				return TRUE
-
-			if(prob(hit.material.getProperty("n_radioactive")*10))
-				hit.AddComponent(/datum/component/radioactive, 50, TRUE, TRUE, 1)
-			if(prob(hit.material.getProperty("radioactive")*10))
-				hit.AddComponent(/datum/component/radioactive, 50, TRUE, FALSE, 1)
-		hit.AddComponent(/datum/component/radioactive, min(O.proj_data.power,25), TRUE, FALSE, 1)
+				var/obj/projectile/reflected = shoot_reflected_bounce(O, hit)
+				reflected.power = O.power
+				return FALSE
+
+			//then fission
+			//fission basically hits like an AoE contamination effect
+			if(hit.material && prob(hit.material.getProperty("n_radioactive")*10))
+				for(var/turf/T in range(1, hit))
+					T.AddComponent(/datum/component/radioactive, 50, TRUE, TRUE, 1)
+			if(hit.material && prob(hit.material.getProperty("radioactive")*10))
+				for(var/turf/T in range(1, hit))
+					T.AddComponent(/datum/component/radioactive, 50, TRUE, FALSE, 1)
+
+			//finally, moderation
+			hit.AddComponent(/datum/component/radioactive, min(O.power, density*multiplier), TRUE, FALSE, 1) //make it all glowy
+			O.power -= density*multiplier
+			if(O.power < 1)
+				O.power = 0
+			return TRUE //don't hit this, lose power and pass through it
+		return TRUE
 
-		if(ismob(hit))
-			var/mob/hitmob = hit
-			hitmob.take_radiation_dose(power/200)
-			O.initial_power = 0
+	tick(var/obj/projectile/P)
+		if(P.power <= 0)
+			P.die()
+			return
 
+	get_power(obj/projectile/P, atom/A)
+		return P.power
```

Comment:
code smell - why does you need this kind of special handling here?

Reply:
because the way projectiles calculate power in `collide()` uses an equation based on `initial_power`, even if you aren't using decay. I'm not using that behaviour and it kinda screws me, so I override.